Mathlib/Data/Complex/Trigonometric.lean

```diff
@@ -0,0 +1,951 @@
+  eq_div_of_mul_eq two_ne_zero <| by
+    rw [cosh, exp, exp, Complex.ofReal_neg, Complex.cosh, mul_two, ← Complex.add_re, ← mul_two,
+      div_mul_cancel₀ _ (two_ne_zero' ℂ), Complex.add_re]
+
+@[simp]
+theorem cosh_zero : cosh 0 = 1 := by simp [cosh]
+
+@[simp]
+theorem cosh_neg : cosh (-x) = cosh x :=
+  ofReal_inj.1 <| by simp
+
+@[simp]
+theorem cosh_abs : cosh |x| = cosh x := by
+  cases le_total x 0 <;> simp [*, _root_.abs_of_nonneg, abs_of_nonpos]
+
+nonrec theorem cosh_add : cosh (x + y) = cosh x * cosh y + sinh x * sinh y := by
+  rw [← ofReal_inj]; simp [cosh_add]
+
+theorem sinh_sub : sinh (x - y) = sinh x * cosh y - cosh x * sinh y := by
+  simp [sub_eq_add_neg, sinh_add, sinh_neg, cosh_neg]
+
+theorem cosh_sub : cosh (x - y) = cosh x * cosh y - sinh x * sinh y := by
+  simp [sub_eq_add_neg, cosh_add, sinh_neg, cosh_neg]
+
+nonrec theorem tanh_eq_sinh_div_cosh : tanh x = sinh x / cosh x :=
+  ofReal_inj.1 <| by simp [tanh_eq_sinh_div_cosh]
+
+@[simp]
+theorem tanh_zero : tanh 0 = 0 := by simp [tanh]
+
+@[simp]
+theorem tanh_neg : tanh (-x) = -tanh x := by simp [tanh, neg_div]
+
+@[simp]
+theorem cosh_add_sinh : cosh x + sinh x = exp x := by rw [← ofReal_inj]; simp
+
+@[simp]
+theorem sinh_add_cosh : sinh x + cosh x = exp x := by rw [add_comm, cosh_add_sinh]
+
+@[simp]
+theorem exp_sub_cosh : exp x - cosh x = sinh x :=
+  sub_eq_iff_eq_add.2 (sinh_add_cosh x).symm
+
+@[simp]
+theorem exp_sub_sinh : exp x - sinh x = cosh x :=
+  sub_eq_iff_eq_add.2 (cosh_add_sinh x).symm
+
+@[simp]
+theorem cosh_sub_sinh : cosh x - sinh x = exp (-x) := by
+  rw [← ofReal_inj]
+  simp
+
+@[simp]
+theorem sinh_sub_cosh : sinh x - cosh x = -exp (-x) := by rw [← neg_sub, cosh_sub_sinh]
+
+@[simp]
+theorem cosh_sq_sub_sinh_sq (x : ℝ) : cosh x ^ 2 - sinh x ^ 2 = 1 := by rw [← ofReal_inj]; simp
+
+nonrec theorem cosh_sq : cosh x ^ 2 = sinh x ^ 2 + 1 := by rw [← ofReal_inj]; simp [cosh_sq]
+
+theorem cosh_sq' : cosh x ^ 2 = 1 + sinh x ^ 2 :=
+  (cosh_sq x).trans (add_comm _ _)
+
+nonrec theorem sinh_sq : sinh x ^ 2 = cosh x ^ 2 - 1 := by rw [← ofReal_inj]; simp [sinh_sq]
+
+nonrec theorem cosh_two_mul : cosh (2 * x) = cosh x ^ 2 + sinh x ^ 2 := by
+  rw [← ofReal_inj]; simp [cosh_two_mul]
+
+nonrec theorem sinh_two_mul : sinh (2 * x) = 2 * sinh x * cosh x := by
+  rw [← ofReal_inj]; simp [sinh_two_mul]
+
+nonrec theorem cosh_three_mul : cosh (3 * x) = 4 * cosh x ^ 3 - 3 * cosh x := by
+  rw [← ofReal_inj]; simp [cosh_three_mul]
+
+nonrec theorem sinh_three_mul : sinh (3 * x) = 4 * sinh x ^ 3 + 3 * sinh x := by
+  rw [← ofReal_inj]; simp [sinh_three_mul]
+
+open IsAbsoluteValue Nat
+
+private theorem add_one_lt_exp_of_pos {x : ℝ} (hx : 0 < x) : x + 1 < exp x :=
+  (by nlinarith : x + 1 < 1 + x + x ^ 2 / 2).trans_le (quadratic_le_exp_of_nonneg hx.le)
+
+private theorem add_one_le_exp_of_nonneg {x : ℝ} (hx : 0 ≤ x) : x + 1 ≤ exp x := by
+  rcases eq_or_lt_of_le hx with (rfl | h)
+  · simp
+  exact (add_one_lt_exp_of_pos h).le
```

Comment:
These results are new, AFAICT. Can you split them into a separate PR, please? Thanks!

Reply:
I didn't add any new results, you can see those lemmas [here](https://github.com/leanprover-community/mathlib4/blob/413249c638b4c9453166d2b94c9a7f13f6238b14/Mathlib/Data/Complex/Exponential.lean#L981) on master. Perhaps the confusion is they are private and this is causing them not to show up in whatever search you are using to look for them?

Reply:
I was using git diff with the `colorMoved` option to check if there was anything but code motion, and these two lemmas were not found. Thanks for explaining, this clarifies it!